rfc/rfc022-discovery-service.md

```diff
@@ -0,0 +1,299 @@
+
+Clients that wish to retain their presence on the list MUST submit a new registration before the current entry's `exp` time has passed.
+
+### 3.4 Retracting presentations
+
+Clients can remove presentations from the list, e.g. when a organization stops supporting the use case.
+To remove presentations of a credential subject the client MUST register a Verifiable Presentation as specified in section 3.1,
+with the following additional requirements:
+
+- it MUST specify `RetractedVerifiablePresentation` as type, in addition to the `VerifiablePresentation`.
+- it MUST contain a `retract_jti` JWT claim, containing the `jti` of the presentation to retract.
+- it MUST NOT contain any credentials.
+
+If a server receives a retraction that references an unknown presentation it MUST respond with a 400 Bad Request response.
+The response MUST be a JSON error response (§3.5) describing the error.
+
+Clients processing a retraction entry MUST remove the presentation indicated by `retract_jti`.
+The expiration of the retraction presentation MUST be equal to the expiration of the presentation to retract.
+
+### 3.5 Error responses
+
+If a server encounters an error while processing a request, it MUST respond with a JSON response according to [RFC7807].
+
+## 4. Presentation processing
+
+The abbreviations below are conform definitions by [IANA](https://www.iana.org/assignments/jwt/jwt.xhtml#claims). 
+To process a presentation, the following validation steps MUST be performed:
+
+- `jti` of the presentation MUST be a non-empty string.
+- `nbf` of the presentation MUST have passed.
+- `exp` of the presentation MUST NOT have passed.
+- `exp` MUST be after `nbf`.
+- `aud` MUST contain the ID of the service.
+- the number of seconds between `nbf` and `exp` MUST NOT exceed `presentation_max_validity` (see Service Definition).
+- all credentials MUST have the same `credentialSubject.id`.
+- `exp` of the presentation MUST NOT be after the expiration date of the credentials.
+- the key used to sign the presentation MUST be owned by the credential subject (see 4.1):
+  the JWT `kid` header MUST reference an `assertionMethod` key from the subject's DID document.
+- The DID of the credential subject MUST be of a DID method defined in the service definition.
+- the presentation MAY contain a `DiscoveryRegistrationCredential` credential defined by the Nuts JSON-LD context (https://nuts.nl/credentials/v1).
+  If present, it MUST NOT be included in the next validation steps.
+- the credentials MUST conform to the Presentation Definition associated with the list (see Service Definition).
+  The Verifiable Presentation MUST NOT contain any Verifiable Credentials besides the ones that conform to the Presentation Definition and the `DiscoveryRegistrationCredential`.
+
+If a validation step fails, the presentation MUST be rejected.
+
+JWT credential and presentation encoding as specified by [VC-DATA-MODEL] MUST be applied.
```

Comment:
is this saying that credentials also MUST be JWT?

Reply:
should only be presentation